File: ts-sdk/src/sdk/models/operations/getarticlebypath.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class GetArticleByPathPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=slug" })
  slug: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=username" })
  username: string;
}


export class GetArticleByPathRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetArticleByPathPathParams;
}


export class GetArticleByPathResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  getArticleByPath200ApplicationJSONObject?: Record<string, any>;
}

File: go-sdk/pkg/models/operations/getarticlebypath.go
package operations

type GetArticleByPathPathParams struct {
	Slug     string `pathParam:"style=simple,explode=false,name=slug"`
	Username string `pathParam:"style=simple,explode=false,name=username"`
}

type GetArticleByPathRequest struct {
	PathParams GetArticleByPathPathParams
}

type GetArticleByPathResponse struct {
	ContentType                              string
	StatusCode                               int64
	GetArticleByPath200ApplicationJSONObject map[string]interface{}
}
